src/build.ts

```diff
@@ -206,6 +206,7 @@ async function generateTransparentAssets(
     }
 
     filePath = resolveTempPngAssetName(filePath)
```

Comment:
Change `resolveTempPngAssetName` signature to be async, then import `mkdir`  from `node:fs/promises` and remove `mkdirSync` from the `node:fs` import.

Then use `mkdir` using `dirname` to get the folder and provide `{ recursive: true }` as options:
```ts
async function resolveTempPngAssetName(name: string) {
  await mkdir(dirname(name), { recursive: true })
  return name.replace(/\.png$/, '-temp.png')
}
```

Add `await` to all `resolveTempPngAssetName` usages.

Finally, remove `checkGenerateFolder` and usages.

This way we add the support also for PWA icons and favicons.

Reply:
ok, resolveTempPngAssetName Changed